src/main/java/com/vaadin/training/bugrap/ui/component/AddCommentPanel.java

```diff
@@ -0,0 +1,66 @@
+package com.vaadin.training.bugrap.ui.component;
+
+import org.vaadin.bugrap.domain.entities.Report;
+import org.vaadin.bugrap.domain.entities.Reporter;
+
+import com.vaadin.training.bugrap.scope.UIScope;
+import com.vaadin.training.bugrap.ui.model.AddCommentApplicationModel;
+import com.vaadin.training.bugrap.ui.model.Models;
+
+public class AddCommentPanel extends AddCommentPanelDesign {
+
+	private static final long serialVersionUID = 101936797233120615L;
+
+	private static final String EMPTY_STRING = "";
+
+	public AddCommentPanel() {
+		initCommentTextArea();
+		initDoneButton();
+		initCancelButton();
+	}
+
+	private void initCancelButton() {
+		cancelButton.addClickListener(event -> {
+			getApplicationModel().reset();
+			clearInputFields();
+		});
+	}
+
+	private void initDoneButton() {
+		doneButton.addClickListener(event -> {
```

Comment:
The done-button could be disabled when comment-field is empty

Reply:
Ok!